flake.nix

```
@@ -249,20 +239,30 @@
             };
           };
 
-          checks.pre-commit-check = pre-commit-check;
-
           devShells = let
-            profillingShell = p: {
+            mkDevShells = p: {
               # `nix develop .#profiling` (or `.#ghc928.profiling): a shell with profiling enabled
               profiling = (p.appendModule {modules = [{enableLibraryProfiling = true;}];}).shell;
+              # `nix develop .#pre-commit` (or `.#ghc928.pre-commit): a shell with pre-commit enabled
+              pre-commit = let
+                pre-commit-check = inputs.pre-commit-hooks.lib.${system}.run {
+                  src = ./.;
+                  hooks = {
+                    fourmolu.enable = true;
+                  };
+                };
+              in
+                p.shell // (nixpkgs.mkShell {
+                shellHook = p.shell.shellHook + pre-commit-check.shellHook;
```

I don't think this does what you want. It seems to leave out the Haskell stuff and have just `pre-commit` in it.

I think you need:

```suggestion
                p.shell.overrideAttrs (old: {
                  shellHook = old.shellHook + pre-commit-check.shellHook;
```